src/view/trip-info-view.js

```diff
@@ -0,0 +1,82 @@
+import AbstractView from '../framework/view/abstract-view.js';
+import {getSumNumbersInArray} from '../utils/common.js';
+import {getTransformationDatesTrip} from '../utils/event-point.js';
+import {sortByDay} from '../utils/event-point.js';
+
+const getCheckedOffersPrice = (offers, eventPoints) => {
+  const checkedOffers = [];
+  for (const point of eventPoints) {
+    const offersByType = offers.find((offer) => offer.type === point.type).offers;
+    checkedOffers.push(offersByType.filter((offer) => point.offers.includes(offer.id)));
+  }
+  const checkedOffersPrice = [];
+  for (const checkedOffer of checkedOffers) {
+    checkedOffersPrice.push(checkedOffer.map((offer) => offer.price));
+  }
+
+  return checkedOffersPrice;
+};
+
+const getTripPrice = (eventPoints, offers) => {
+  const eventPointsPrices = eventPoints.map((point) => Number(point.basePrice));
+  const generalBasePrice = getSumNumbersInArray(eventPointsPrices);
+
+  const checkedOffersPriceSum = getSumNumbersInArray(getCheckedOffersPrice(offers, eventPoints).map((item) => getSumNumbersInArray(item)));
+
+  return generalBasePrice + checkedOffersPriceSum;
+};
+
+const getTripTitle = (eventPoints, destinations) => {
+  const sorteredPoints = eventPoints.sort(sortByDay);
+  const getDestinationById = (dests, id) => dests.find((dest) => dest.id === id);
+  const tripNames = [];
+  for (const point of sorteredPoints) {
+    tripNames.push(getDestinationById(destinations, point.destination).name);
+  }
+
+  return tripNames.length > 3 ? `
+  ${tripNames[0]}&nbsp;&mdash;&nbsp;&hellip;&nbsp;&mdash;&nbsp;${tripNames[tripNames.length - 1]}
```

Comment:
Не очень читаемо конечно выглядит) Но можно оставить, также обрати внимание на новый метод at который позволяет брать элементы с конца
```suggestion
  ${tripNames[0]}&nbsp;&mdash;&nbsp;&hellip;&nbsp;&mdash;&nbsp;${tripNames.at(-1)}
```